src/Commands/ConvertToDrupalRecommendedSiteCommand.php

```
@@ -103,13 +105,19 @@ public function convert(
             );
             $this->log()->notice('composer.json updated to match "drupal-recommended" upstream');
         } catch (Throwable $t) {
+            $errors++;
```

@kporras07 `$errors` is not needed, ie:
```php
// ...
} catch (Throwable $t) {
    $this->log()->error(
        sprintf(
            'Failed updating composer.json: %s',
            $t->getMessage()
        )
    );
     throw new ComposerException(
        'Failed updating composer.json. Please check the logs for more information.'
    );
}
```

---

As per the ticket, we want to let the current set of composer command to run and then fail with the exception. That's why I was writing it like that. Please let me know if you think your suggestion is still applicable having this bit of context

---

The error counting needs to happen inside the `foreach`, though; in its current location, the first exception thrown will still abort the entire loop.

---

@greg-1-anderson I'm not sure I'm following. We catch throwable and increment errors so that after the try-catch block we throw ComposerException that will actually stop the process. Am I missing something?

---

Yes; $errors will only ever be zero or 1, because if some of the code inside `foreach ($drupalRecommendedComposerDependencies as $dependency) {` throws, the loop will not continue, because the `try` is outside the loop instead of inside.

---

ohhhh! That makes sense. Thanks!

---

I think that the outer try-catch can just go away now. Let it throw all the way out of the command if something outside the inner loop fails.

---

Done!